woocommerce/woocommerce-gateway-stripe
# Payment method customization

This PR makes some further implementation of the same settings page on https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2750

## Changes proposed in this Pull Request:
- Introduces separate update endpoint for payment method settings. 
- Added necessary actions, selectors and hooks for updating individual payment method settings.
- The changes are saved to the backend when `Save changes` button is clicked now.
- Allows customizing the `card` method also as per design.
- Added the `Expiration` field for the methods that have `expiration` option in settings (Boleto is the only one with this option)

![customization section](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/5692d708-2beb-4e9c-899e-089ef2e59686)


## Testing instructions
- Go to the Stripe settings page.
- From the `Settings` tab disable UPE. The customization button for each method is available only when UPE is disabled.
- Go to the `Payment Methods` tab. Click the customize button beside any payment method.
- Add a name and description. Click `Save changes`. In the network tab there should be a successful POST request to `<your_site>/wp-json/wc/v3/wc_stripe/settings/payment_method` and the changes should be saved.
